sql/src/main/java/io/crate/operation/udf/UserDefinedFunctionService.java

```diff
@@ -109,6 +112,36 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
             });
     }
 
+    void dropFunction(final DropUserDefinedFunctionRequest request,
+                      final ActionListener<ClusterStateUpdateResponse> listener) {
+        if (!clusterService.localNode().isMasterNode()) {
```

Comment:
This is only checking if this node *could* be the master, not if it currently *is* the master.

Need to use something like this:

```
        DiscoveryNodes nodes = state.nodes();
        if (nodes.getMasterNodeId().equals(nodes.getLocalNodeId()) == false) {
```